Subject: [PATCH] Fix i8259 for target-arches not supporting KVM
From: Sergei Trofimovich <slyfox@xxxxxxxx>
Date: Sun, 29 Jan 2012 16:18:01 +0300

  $ ./configure --target-list=alpha-softmmu && make

      CC    alpha-softmmu/i8259.o
    cc1: warnings being treated as errors
    /home/slyfox/dev/git/qemu-kvm/hw/i8259.c: In function 'kvm_i8259_set_irq':
    /home/slyfox/dev/git/qemu-kvm/hw/i8259.c:688:13: error: implicit declaration of function 'apic_set_irq_delivered'

    /home/slyfox/dev/git/qemu-kvm/hw/i8259.c:688:13: error: nested extern declaration of 'apic_set_irq_delivered'
    make[1]: *** [i8259.o] Error 1
    make: *** [subdir-alpha-softmmu] Error 2

      LINK  alpha-softmmu/qemu-system-alpha
    i8259.o: In function `kvm_i8259_set_irq':
    /tmp/portage/app-emulation/qemu-kvm-9999/work/qemu-kvm-9999/hw/i8259.c:689: undefined reference to `apic_set_irq_delivered'

Signed-off-by: Sergei Trofimovich <slyfox@xxxxxxxxxx>
---
 hw/i8259.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/i8259.c b/hw/i8259.c
index 0632ea2..2f6789d 100644
--- a/hw/i8259.c
+++ b/hw/i8259.c
@@ -21,6 +21,7 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
+#include "apic.h"
 #include "hw.h"
 #include "pc.h"
 #include "isa.h"
@@ -682,12 +683,14 @@ static int kvm_kernel_pic_load_from_user(PicState *s)
 
 static void kvm_i8259_set_irq(void *opaque, int irq, int level)
 {
+#ifdef CONFIG_KVM
     int pic_ret;
     if (kvm_set_irq(irq, level, &pic_ret)) {
         if (pic_ret != 0)
             apic_set_irq_delivered();
         return;
     }
+#endif
 }
 
 device_init(pic_register)
-- 
1.7.8.3
